lib/rspec/rails/matchers/active_job.rb

```diff
@@ -26,7 +27,12 @@ def matches?(proc)
             in_block_jobs = queue_adapter.enqueued_jobs.drop(original_enqueued_jobs_count)
 
             @matching_jobs_count = in_block_jobs.count do |job|
-              serialized_attributes.all? { |key, value| value == job[key] }
+              if serialized_attributes.all? { |key, value| value == job[key] }
+                @block.call(*job[:args])
```

Comment:
My thinking was that it could be useful to try to disambiguate where the block is intended to be applied to.  For instance if multiple jobs get enqueued, you may only want to check the arguments of one of them.  E.g. calling the block always would break this spec:

https://github.com/jschroeder9000/rspec-rails/blob/8383c95f244168eaaf9deb1553b96d5a3e06ec81/spec/rspec/rails/matchers/active_job_spec.rb#L244

But if that spec were called by passing paramaters to `with` instead of a block, it would pass.  I think they should be consistent.

Giving it some more thought, though, that does seem like a flawed approach because there still would have to be some other unique attributes about the jobs to differentiate them and even that may not always be the case.

In the simplest and most general case, I think it makes sense both of these specs should behave identically:

```ruby
it 'passes with arguments' do
  expect {
    hello_job.perform_later('asdf')
    hello_job.perform_later('zxcv')
  }.to have_enqueued_job(hello_job).with('asdf')
end

it 'passes with a block' do
  expect {
    hello_job.perform_later('asdf')
    hello_job.perform_later('zxcv')
  }.to have_enqueued_job(hello_job).with { |arg|
    expect(arg).to eq('asdf')
  }
end
```

The behavior of passing arguments is that the spec would pass, but if I just always call the block without doing anything else then it would fail because _all_ enqueued jobs would have to meet the expectations of the block.  But that is inconsistent with the behavior of passing arguments.

Maybe a better approach is to call the block in a begin block and if it doesn't raise an ExpectationNotMetError then it passes.  Something like:

```ruby
@matching_jobs_count = in_block_jobs.count do |job|
  if serialized_attributes.all? { |key, value| value == job[key] }
    block_passes = true
    begin
      @block.call(*job[:args])
    rescue RSpec::Expectations::ExpectationNotMetError
      block_passes = false
    end
    block_passes
  else
    false
  end
end
```

That would keep the behavior of both cases consistent.  The behavior of passing arguments is already to tell you that the count of matched enqueued jobs did not match what is expected and it does not tell you what differed from the jobs that were enqueued, so no information is lost by suppressing the ExpectationNotMetError when passing a block versus passing arguments.

What do you think?